server/src/main/scala/org/http4s/server/middleware/RequestLogger.scala

```diff
@@ -42,15 +42,35 @@ object RequestLogger {
             )
 
             val response = routes(changedRequest)
-            response.map { resp =>
-              resp.withBodyStream(
-                resp.body.onFinalize(
+            response.attempt
+              .flatMap {
+                case Left(e) =>
+                  OptionT.liftF(
```

Comment:
I think this entire thing can just be one big `OptionT.liftF(logyourbleepblorp >> F.raiseError(e))`  if we want to be pedantic.